src/app/profiles/(components)/MultipleChoiceComponent.tsx

```diff
@@ -0,0 +1,162 @@
+import { Input } from "@/components/ui/input"
+
+type Answer = {
+    id: number
+    label: string
+}
+
+const INITIAL_QUESTION: string = "What is your favorite color?"
+const INITIAL_ANSWERS: Answer[] = [
+    {id: 1, label: ""},
+]
+const INITIAL_VALUES = {
+    question: INITIAL_QUESTION,
+    answers: INITIAL_ANSWERS,
+}
+
+const SortableInput = ({index}: {index: number}) => {
+    const { values, handleChange, setFieldValue } = useFormikContext<{
+        question: string;
+        answers: Answer[];
+    }>()
+
+    const { attributes, listeners, setNodeRef, transform, transition } = useSortable({id: values.answers[index].id})
+    const style = {
+        transition,
+        transform: CSS.Transform.toString(transform),
+    }
+
+    return (
+        <div style={style} {...attributes}>
+            <br></br>
+            <Label>Answer {index + 1}</Label>
+            <div className="grid grid-cols-10 gap-3">
+                { values.answers.length > 1 ?
+                    <>
+                        <Input className="col-span-8" name={`answers.${index}.label`} type="text"  onChange={handleChange} />
+                        <Button type="button" className="col-span-1" onClick={() => (setFieldValue('answers', values.answers.filter((_, i: number) => i !== index)))}>
+                            <Trash />
+                        </Button>
+                        <button type="button" className="col-span-1 content-center" ref={setNodeRef} {...listeners}>
+                            <Drag />
+                        </button>
+                    </>
+                    :
+                    <Input className="col-span-10" name={`answers.${index}.label`} type="text" onChange={handleChange} />
+                }
+
+            </div>
+        </div>
+    )
+}
+
+const MultipleChoiceEditor = () => {
+    const { values, setFieldValue, handleChange } = useFormikContext<{
+        question: string;
+        answers: Answer[];
+    }>()
+
+    const onDragEnd = (event: DragEndEvent) => {
+        const {active, over} = event
+        if (over && active.id === over.id) {
+            return
+        }
+
+        const oldIndex = values.answers.findIndex((answer) => answer.id === active.id)
+        const newIndex = over ? values.answers.findIndex((answer) => answer.id === over.id) : -1
+
+        setFieldValue('answers', arrayMove(values.answers, oldIndex, newIndex))
+    }
+
+    return (
+        <Card>
+            <Form>
+                <CardHeader>
+                    <CardTitle>Multiple Choice Question</CardTitle>
+                    <CardDescription>Order of the answers is tiered and the top one is considered higher/better</CardDescription>
+                </CardHeader>
+                <CardContent>
+                    <Label>Question Text</Label>
+                    <Input name="question" type="text" onChange={handleChange} placeholder="Question Text" defaultValue={INITIAL_QUESTION} />
```

Comment:
I don't think we need `defaultValue` here, it's not going to be able to work like this once you have to actually use real data here.

If you 100% must define it, I think there's a way to get it from formik's context